core/lib/instrument_api.h

```diff
@@ -1676,6 +1676,16 @@ dr_standalone_exit(void);
  * UTILITY ROUTINES
  */
 
+#    ifdef UNIX
+DR_API
+/**
+ * Forge (emulate) OS signal with a given number at the specified PC.
+ * \note Only Unix.
```

Comment:
Please document that this function does not return when successful (maybe return a bool like dr_redirect_execution?).  That means it cannot be called from random places where a lock might be held or something, which also needs to be documented: probably using the same limits as dr_redirect_execution() would be safe.

Reply:
The documentation for `dr_redirect_execution` does not specify any particular limitations related to no return if the call is successful. I will describe that this function is not supposed to return and that care should be taken to avoid any stale state or unreleased resources.

Reply:
The documentation for dr_redirect_execution says explicitly "if successful, does not return" (the last sentence in the docs: https://dynamorio.org/dynamorio_docs/dr__ir__utils_8h.html#a9f612cf9d0ca2025b23d30b6f5bb0231)

Reply:
OK I missed the context (email just shows yours, and my initial comment above is from a month ago which I have forgotten): you're replying to limitations on when dr_redirect_execution can be called.  The docs explicitly say it can only be called from a clean call or a Windows exception event.